### PR TITLE
[receiver/splunkentperisereceiver] Add Dispatch Artifact Collection

### PR DESCRIPTION
**Pull request by michael-burt**
#### Description

This PR adds metrics for search dispatch using the `/services/server/status/dispatch-artifacts` API.

#### Testing

I added unit tests for these metrics

### CODE REVIEW

File: receiver/splunkenterprisereceiver/scraper.go [hunk 1]
```diff
@@ -1627,3 +1628,96 @@ func (s *splunkScraper) scrapeKVStoreStatus(ctx context.Context, now pcommon.Tim
...
+		}
+
+		if !s.conf.MetricsBuilderConfig.Metrics.SplunkServerSearchesInvalidTotal.Enabled {
+			invalidCount, err := strconv.ParseInt(f.Content.InvalidCount, 10, 64)
+			if err != nil {
+				errs <- err
+				return
+			}
+			s.mb.RecordSplunkServerSearchesInvalidTotalDataPoint(now, invalidCount, s.conf.SHEndpoint.Endpoint)
+		}
```
Comment (lhan-splunk) [inline receiver/splunkenterprisereceiver/scraper.go:1721]: add

		if !s.conf.MetricsBuilderConfig.Metrics.SplunkServerSearchesSavedSearchTotal.Enabled {
			invalidCount, err := strconv.ParseInt(f.Content.SavedSearchCount, 10, 64)
			if err != nil {
				errs <- err
				return
			}
			s.mb.RecordSplunkServerSearchesSavedSearchTotalDataPoint(now, savedSearchesCount, s.conf.SHEndpoint.Endpoint)
		}

File: receiver/splunkenterprisereceiver/metadata.yaml [hunk 2]
```diff
@@ -268,7 +268,62 @@ metrics:
     unit: '{status}'
     gauge:
       value_type: int
-    attributes: [splunk.kvstore.status.value]
+    attributes: [splunk.kvstore.status.value] 
+  #'services/server/status/dispatch-artifacts'
+  splunk.server.searches.total:
```
Comment (lhan-splunk) [inline receiver/splunkenterprisereceiver/metadata.yaml:273]: remove line 273-281

File: receiver/splunkenterprisereceiver/metadata.yaml [hunk 3]
```diff
@@ -268,7 +268,62 @@ metrics:
...
+    description: Counter tracking number of dispatched search jobs since start-up. *Note:** Must be pointed at specific Search Head `endpoint` and gathers metrics from only that Search Head.
+    unit: "{searches}"
+    sum:
+      monotonic: true
+      aggregation_temporality: cumulative
+      value_type: int
+    attributes: [splunk.host]
+  splunk.server.searches.adhoc.total:
+    enabled: false
+    description: Counter tracking number of ad hoc search jobs since start-up. *Note:** Must be pointed at specific Search Head `endpoint` and gathers metrics from only that Search Head.
```
Comment (lhan-splunk) [inline receiver/splunkenterprisereceiver/metadata.yaml:284]: description: Counter tracking number of ad hoc search artifacts currently on the disk. *Note:** Must be pointed at specific Search Head `endpoint` and gathers metrics from only that Search Head.

File: receiver/splunkenterprisereceiver/metadata.yaml [hunk 4]
```diff
@@ -268,7 +268,62 @@ metrics:
...
+    unit: "{searches}"
+    sum:
+      monotonic: true
+      aggregation_temporality: cumulative
+      value_type: int
+    attributes: [splunk.host]
+  splunk.server.searches.scheduled.total:
+    enabled: false
+    description: Counter tracking number of scheduled search jobs since start-up. *Note:** Must be pointed at specific Search Head `endpoint` and gathers metrics from only that Search Head.
```
Comment (lhan-splunk) [inline receiver/splunkenterprisereceiver/metadata.yaml:293]: description: Counter tracking number of scheduled search artifacts currently on the disk. *Note:** Must be pointed at specific Search Head `endpoint` and gathers metrics from only that Search Head.

File: receiver/splunkenterprisereceiver/metadata.yaml [hunk 5]
```diff
@@ -268,7 +268,62 @@ metrics:
...
+    unit: "{searches}"
+    sum:
+      monotonic: true
+      aggregation_temporality: cumulative
+      value_type: int
+    attributes: [splunk.host]
+  splunk.server.searches.completed.total:
+    enabled: false
+    description: Counter tracking number of completed search jobs since start-up. *Note:** Must be pointed at specific Search Head `endpoint` and gathers metrics from only that Search Head.
```
Comment (lhan-splunk) [inline receiver/splunkenterprisereceiver/metadata.yaml:302]: description: Counter tracking number of artifacts currently on the disk that belong to finished searches. Note:* Must be pointed at specific Search Head endpoint and gathers metrics from only that Search Head.

File: receiver/splunkenterprisereceiver/metadata.yaml [hunk 6]
```diff
@@ -268,7 +268,62 @@ metrics:
...
+    unit: "{searches}"
+    sum:
+      monotonic: true
+      aggregation_temporality: cumulative
+      value_type: int
+    attributes: [splunk.host]
+  splunk.server.searches.incomplete.total:
+    enabled: false
+    description: Counter tracking number of incomplete search jobs since start-up. *Note:** Must be pointed at specific Search Head `endpoint` and gathers metrics from only that Search Head.
```
Comment (lhan-splunk) [inline receiver/splunkenterprisereceiver/metadata.yaml:311]: description: Counter tracking number of artifacts currently on the disk that belong to unfinished/running searches. Note:* Must be pointed at specific Search Head endpoint and gathers metrics from only that Search Head.

File: receiver/splunkenterprisereceiver/metadata.yaml [hunk 7]
```diff
@@ -268,7 +268,62 @@ metrics:
...
+    unit: "{searches}"
+    sum:
+      monotonic: true
+      aggregation_temporality: cumulative
+      value_type: int
+    attributes: [splunk.host]
+  splunk.server.searches.invalid.total:
+    enabled: false
+    description: Counter tracking number of invalid search jobs since start-up. *Note:** Must be pointed at specific Search Head `endpoint` and gathers metrics from only that Search Head.
```
Comment (lhan-splunk) [inline receiver/splunkenterprisereceiver/metadata.yaml:320]: description: Counter tracking number of artifacts currently on the disk that are not in a valid state, such as missing  info.csv file, etc. Note:* Must be pointed at specific Search Head endpoint and gathers metrics from only that Search Head.

File: receiver/splunkenterprisereceiver/metadata.yaml [hunk 8]
```diff
@@ -268,7 +268,62 @@ metrics:
...
+    unit: "{searches}"
+    sum:
+      monotonic: true
+      aggregation_temporality: cumulative
+      value_type: int
+    attributes: [splunk.host]
```
Comment (lhan-splunk) [inline receiver/splunkenterprisereceiver/metadata.yaml:326]: Add

  splunk.server.searches.savedSearches.total:
    enabled: false
    description: Counter tracking, for the splunk.server.searches.scheduled.total number of scheduled search artifacts,  how many different saved-search they belong to. *Note:** Must be pointed at specific Search Head `endpoint` and gathers metrics from only that Search Head.
    unit: "{searches}"
    sum:
      monotonic: true
      aggregation_temporality: cumulative
      value_type: int
    attributes: [splunk.host]

**Review by MovieStoreGuy — APPROVED** [review]

**Pull request merged**